applications/dashboard/settings/class.hooks.php

```diff
@@ -656,6 +658,23 @@ public function gdn_dispatcher_appStartup_handler($sender) {
         $this->checkAccessToken();
     }
 
+    /**
+     * Check to see if a user is banned.
+     *
+     * @throws Exception if the user is banned.
+     */
+    public function base_afterSignIn_handler() {
+        if ($ban = Gdn::session()->getPermissions()->getBan([])) {
```

Comment:
I think what we're trying to target here are user rows with their `Banned` or `Deleted` flags toggled on. This is going to trip for every ban type (e.g. 2FA, CSRF). Wouldn't it be better to test for just the ban types you're expecting? Maybe someone signs into the site and is "banned" (no permissions) until they enter their 2FA auth.

The empty array in the call to `getBan` is superfluous, because the default is already an empty array.

Reply:
I'm pretty sure that all ban reasons should fail here.

Reply:
It's my understanding that we just want to block deleted and banned users from accessing the site here.

My take on permission bans is that they're meant to strip user permissions, not necessarily block them from signing in. There's several ban types that aren't necessarily related to signing in (2FA, CSRF, updating, uninstall, private community, etc.). Addons can pretty much do whatever they like there. I think it's maybe casting the net too wide to reject anyone from signing into the site who has any kind of permission "ban". I don't think "you don't have any permissions" should always equate to "you cannot sign in".

Reply:
Makes sense. I updated the code appropriately.